graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/QueryFetcher.java

```diff
@@ -93,16 +97,63 @@ private BoundQuery buildQuery(DataFetchingEnvironment environment, DataStore dat
         limit = (int) limitObj;
       }
     }
-    return dataStore
-        .queryBuilder()
-        .select()
-        .column(buildQueryColumns(environment))
-        .from(table.keyspace(), table.name())
-        .where(buildClause(table, environment))
-        .limit(limit)
-        .orderBy(buildOrderBy(environment))
-        .build()
-        .bind();
+    QueryBuilder.QueryBuilder__41 queryBuilder =
+        dataStore
+            .queryBuilder()
+            .select()
+            .column(buildQueryColumns(environment))
+            .from(table.keyspace(), table.name())
+            .where(buildClause(table, environment))
+            .limit(limit)
+            .orderBy(buildOrderBy(environment));
+
+    addAggregationFunctions(environment, queryBuilder);
+
+    return queryBuilder.build().bind();
+  }
+
+  private void addAggregationFunctions(
+      DataFetchingEnvironment environment, QueryBuilder.QueryBuilder__41 queryBuilder) {
+    List<SelectedField> valuesFields = environment.getSelectionSet().getFields("values");
+    if (valuesFields.isEmpty()) {
+      return;
+    }
+    for (SelectedField valuesField : valuesFields) {
+      for (SelectedField selectedField : valuesField.getSelectionSet().getFields()) {
+        if (selectedField.getName().equals(INT_FUNCTION)) {
+          Map<String, Object> arguments = selectedField.getArguments();
+          String functionName = (String) arguments.get("name");
+          if (functionName.equals(COUNT)) {
+            addCount(arguments, queryBuilder, selectedField);
+          }
+        }
+      }
+    }
+  }
+
+  private void addCount(
+      Map<String, Object> arguments,
+      QueryBuilder.QueryBuilder__41 queryBuilder,
+      SelectedField selectedField) {
+    @SuppressWarnings("unchecked")
+    List<String> args = (List<String>) arguments.get("args");
+    if (args.size() != 1) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The %s function takes only one argument, " + "but more arguments: %s were provided.",
+              COUNT, args));
+    }
+    String column = getDBColumnName(table, args.get(0));
+    if (column == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "The column name: %s provided for the %s function does not exists.",
+              args.get(0), COUNT));
+    }
+
+    // todo after https://github.com/stargate/stargate/issues/907 is done
```

Comment:
we need https://github.com/stargate/stargate/issues/907 to have aggregations available on the builder.